ale_linters/elm/make.vim

```diff
@@ -3,20 +3,31 @@
 
 function! ale_linters#elm#make#Handle(buffer, lines)
     let l:output = []
+    let l:is_windows = has('win32')
+    let l:temp_dir = l:is_windows ? $TMP : $TMPDIR
     for l:line in a:lines
         if l:line[0] ==# '['
             let l:errors = json_decode(l:line)
 
             for l:error in l:errors
-                call add(l:output, {
-                \    'bufnr': a:buffer,
-                \    'lnum': l:error.region.start.line,
-                \    'vcol': 0,
-                \    'col': l:error.region.start.column,
-                \    'type': (l:error.type ==? 'error') ? 'E' : 'W',
-                \    'text': l:error.overview,
-                \    'nr': -1,
-                \})
+                " Check if file is from the temp directory.
+                " Filters out any errors not related to the buffer.
+                if l:is_windows
+                    let l:file_is_buffer = l:error.file[0:len(l:temp_dir)-1] ==? l:temp_dir
+                else
+                    let l:file_is_buffer = l:error.file[0:len(l:temp_dir)-1] ==# l:temp_dir
+                endif
+                if l:error.file[0:len(l:temp_dir)-1] == l:temp_dir
```

Comment:
I think you forgot to use the variable you created above here. Also, could you put a single blank line above if statements which don't start a new block?